File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> my_dict = new HashMap<>();
        for (int i = 0; i <= nums.length; i++){
            int comp = target - nums[i];
            if (my_dict.containsKey(comp)){
                return new int[]{my_dict.get(comp), i};
            }
            my_dict.put(nums[i], i);
        }
        return new int[0];
        
    }
}

File: 0001-two-sum/0001-two-sum.cpp
#include <vector>
#include <unordered_map>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> my_dict;
        for (int i=0; i <= nums.size(); i++){
            int comp = target - nums[i];
            if (my_dict.find(comp) != my_dict.end()){
                  return {my_dict[comp], i};

            }
            my_dict[nums[i]] = i;

        }
        return {};          
    }    
};